at_client/connections/atmonitorconnection.py

```diff
@@ -0,0 +1,222 @@
+                        event_type = AtEventType.MONITOR_EXCEPTION
+                        event_data["key"] = "__monitorException__"
+                        event_data["value"] = response.decode()
+                        event_data["exception"] = "Unexpected 'data:' message from server"
+                    elif response.startswith(b"error:"):
+                        event_type = AtEventType.MONITOR_EXCEPTION
+                        event_data["key"] = "__monitorException__"
+                        event_data["value"] = response.decode()
+                        event_data["exception"] = "Unexpected 'error:' message from server"
+                    elif response.startswith(b"notification:"):
+                        event_data = json.loads(response.decode()[len("notification:"):])
+                        uuid = str(event_data.get("id"))
+                        operation = str(event_data.get("operation"))
+                        key = str(event_data.get("key"))
+                        if "epochMillis" in event_data:
+                            self.last_received_time = int(event_data.get("epochMillis"))
+                        else:
+                            self.last_received_time = TimeUtil.current_time_millis()
+                        if uuid == "-1":
+                            event_type = AtEventType.STATS_NOTIFICATION
+                        elif operation == "update":
+                            if key.startswith(str(self.atsign.to_string) + ":shared_key@"):
+                                event_type = AtEventType.SHARED_KEY_NOTIFICATION
+                            else:
+                                event_type = AtEventType.UPDATE_NOTIFICATION
+                        elif operation == "delete":
+                            event_type = AtEventType.DELETE_NOTIFICATION
+                        else:
+                            event_type = AtEventType.MONITOR_EXCEPTION
+                            event_data["key"] = "__monitorException__"
+                            event_data["value"] = response.decode()
+                            event_data["exception"] = "Unknown notification operation " + str(operation)
+                    else:
+                        event_type = AtEventType.MONITOR_EXCEPTION
+                        event_data["key"] = "__monitorException__"
+                        event_data["value"] = response.decode()
+                        event_data["exception"] = "Malformed response from server"
+                except Exception as e:
+                    print(e)
+                    event_type = AtEventType.MONITOR_EXCEPTION
+                    event_data["key"] = "__monitorException__"
+                    event_data["value"] = response.decode()
+                    event_data["exception"] = str(e)
+                    traceback.print_exc()
+
+                at_event = AtEvent(event_type, event_data)
+                self.queue.put(at_event)
```

Comment:
self.queue.put is a coroutine, should be awaited
```       
await self.queue.put(at_event)
```
But since _run isn't async maybe add something like this:
```
async def _put(self, at_event:AtEvent):
        """ Put an event into the queue"""
        await self.queue.put(at_event)
```


Or we can consider using this, which doesn't wait to put but instead throws a QueueFull exception
```
self.queue.put_nowait()
```

Reply:
Be careful, that runs in its own thread. It shouldn't be a problem. It works nicely for REPL code at least